### PR TITLE
Add logout button to access denied page

### PR DESCRIPTION
**Pull request by RubenVerborgh**
Unfortunately it doesn't seem to work because server-side logout does not seem to work (WebID is not deleted from session).

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
@kidehen Is logout working for you?

**Comment by RubenVerborgh** [comment]
Blocked by https://github.com/solid/oidc-rp/issues/21

File: default-views/auth/no-permission.hbs [hunk 1]
```diff
@@ -17,8 +17,18 @@
       You are currently logged in as <code>{{webId}}</code>,
       but do not have permission to access <code>{{currentUrl}}</code>.
     </p>
+    <p>
+      <button class="btn btn-primary" onclick="logout()">Log out</button>
```
Comment (rubensworks) [inline default-views/auth/no-permission.hbs]: Perhaps we can use `btn-danger` here instead, for consistency with the other logout buttons.

Comment (kjetilk) [reply default-views/auth/no-permission.hbs]: Consistency is a good thing. :-)

**Comment by RubenVerborgh** [comment]
Unblocked, new reviews welcome.

**Comment by RubenVerborgh** [comment]
Improvement coming 🙂

**Comment by RubenVerborgh** [comment]
Now with d7a1c0bb6bdfd0e52fcd4e74e9a74d8efb0c1ceb, which ensures that all pages uses the same login/logout code. This a) removes duplication b) ensures the latest code (since we cannot change home pages once people have created them) and c) ensures that logging in on the homepage is materialized into a cookie and thus carried over to other pages.

**Comment by megoth** [comment]
> @megoth And now even better with the latest commit, removing `onclick` as remarked by @csarven.

Yeah, I'm a bit on the fence there, tbh; I'm kinda enjoying onclick as it makes it much easier to debug later on. But it does require adding methods to the global scope, which is A Bad Thing [tm] in general (for various reasons we don't need to get into here). In this case it's a small application, so it's probably better to use addEventListener (for bigger applications I recommend having a convention to handlers addEventListener makes use of, e.g. prefix `js-` for CSS classes).

But this won't be a problem with applications like React, so I'll drop my rant here =P

**Comment by RubenVerborgh** [comment]
> But it does require adding methods to the global scope

And also, it would tie homepages to a specific version of the auth-button script, which would prevent us from updating that script in the future. (Related: #936.)

**Pull request merged**